api/metrics/v1/rules_postgres.go

```diff
@@ -0,0 +1,157 @@
+package v1
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/rulefmt"
+	"gopkg.in/yaml.v3"
+)
+
+// Statically check that the Postgres rules repository implements the RulesRepository interface.
+var _ RulesRepository = (*RulesRepositoryPostgres)(nil)
+
+// NewRulesRepository implements RulesRepository interface with Postgres
+// which the RulesAPI requires.
+func NewRulesRepository(db *sql.DB) *RulesRepositoryPostgres {
+	return &RulesRepositoryPostgres{db: db}
+}
+
+// RulesRepositoryPostgres the actual implementation of RulesRepository for Postgres.
+type RulesRepositoryPostgres struct {
+	db *sql.DB
+}
+
+func (r *RulesRepositoryPostgres) ListRuleGroups(ctx context.Context, tenant string) (RuleGroups, error) {
+	var groups RuleGroups
+
+	query := `SELECT name, interval, rules FROM metrics_rules WHERE tenant = $1 ORDER BY name ASC`
```

Comment:
I'm assuming then that the database has been spun up? The design doc still says "some sort of database - and thus we can for now only persist to disk" so make me think these code paths are unused?

Reply:
We are totally not in the business of spinning up DBs. If you want to run Observatorium with a rules repository that is rea/write then you need to provide a Postgres DB. All operations will then create/read/update/delete rows in this DB table. This code path _is_ used. The e2e tests that are running in this PR do the following:
1. spin up CockroachDB
2. run a migration to create the necessary table in the DB
3. create/update/read rules from the DB using the API

Reply:
This PR also enables another read-only rules repository: every tenant can specify their own file that holds the rules for their users. When running like this, users can only read from the file and cannot create, update, or delete rules.

Reply:
I think we should chat in person about this :) Apologies, but I certainty don't feel like I have enough context to approve this PR, as I do not fully comprehend the use-cases or implications of merging this :raised_hands: